ledger/metrics/src/main/scala/com/daml/telemetry/Telemetry.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.telemetry
+
+import java.util.{Map => jMap}
+
+import io.opentelemetry.api.trace.Span
+
+import scala.concurrent.Future
+
+trait Telemetry {
+
+  /** Returns a telemetry context from the OpenTelemetry context stored in the gRPC
+    * thread local context.
+    * This is used to recover the tracing metadata from an incoming gRPC call,
+    * and used for the subsequent spans.
+    *
+    * Current implementation only works with OpenTelemetry < 0.10. Later versions of
+    * OpenTelemetry use a different context object.
+    */
+  def contextFromGrpcThreadLocalContext(): TelemetryContext
+
+  /** Returns a telemetry context from the metadata encoded in the given key-value map.
+    *
+    * Typically, the metadata map has been encoded in a message, which has been sent
+    * across boundaries, from a different process.
+    * Originally, it has been created to carry tracing metadata across boundaries, and
+    * to create complete traces.
+    *
+    * @see [[com.daml.telemetry.TelemetryContext.encodeMetadata()]]
+    */
+  def contextFromMetadata(metadata: Option[jMap[String, String]]): TelemetryContext
```

Comment:
There are a few methods like this one, `TelemetryContext.runInOpenTelemetryScope` and `TelemetryContext.encodeMetadata` that are not used in this repos, shall I move them back somehow to the oem-integration-kit repo?